src/internal/superjson/mutate.ts

```diff
@@ -257,32 +277,60 @@ export function addProfileProvider(
       file: trimFileURI(payload),
       defaults: profileProvider.defaults,
     };
+    targetedProfile.priority = resolvePriorityAddition(
+      targetedProfile.priority,
+      providerName
+    );
 
     return true;
   }
 
   // Priority #2: keep previous structure and merge
   let defaults: ProfileProviderDefaults | undefined;
   if (typeof profileProvider === 'string') {
+    //Change
     defaults = payload.defaults;
-  } else if (profileProvider.defaults) {
-    if (!payload.defaults) {
-      defaults = targetedProfile.defaults;
-    } else {
+  } else {
+    if (profileProvider.defaults && payload.defaults) {
+      //Chanage
       //Merge existing with new
       defaults = mergeVariables(
         castToNonPrimitive(profileProvider.defaults) || {},
         castToNonPrimitive(payload.defaults) || {}
       ) as ProfileProviderDefaults;
+    } else if (!profileProvider.defaults && payload.defaults) {
+      defaults = payload.defaults;
     }
   }
+  // if there no other keys and we changed defaults
+  if (
+    !('file' in payload) &&
+    !('mapVariant' in payload) &&
+    !('mapRevision' in payload) &&
+    defaults
+  ) {
+    targetedProfile.providers[providerName] = {
+      ...(typeof profileProvider === 'string' ? {} : profileProvider),
```

Comment:
Don't we need to preserve it as a `file` entry?

Reply:
You mean like:
if typeof profileProvider === 'string' {
targetedProfile.providers[providerName] = {
file: profileProvider
defaults: payload.defaults
}
}

Reply:
`    targetedProfile.providers[providerName] = {
      ...(typeof profileProvider === 'string' ? { file: profileProvider } : profileProvider),
      defaults,
    };`

Reply:
Yeah, otherwise I think we'd lose the path